### PR TITLE
Implement MessageSendForbidden error and handling

### PR DESCRIPTION
**Pull request by MattyTheHacker**

### CODE REVIEW

**Review by CarrotManMatt — CHANGES_REQUESTED** [review]
I like the approach taken here, however have 2 main changes that need to be made (before I make comments on specific formatting changes).

1. The `@capture_guild_does_not_exist_error` decorator (and other error capture decorators) is/are used for any interactions that do not have exceptions captured by default (I.e. tasks & event listeners need exceptions capturing, whereas commands have exceptions captured by the `on_application_command_error` event listener). Therefore handling of capturing the `MessageSendForbiddenError` should be performed by the decorators for the tasks & event listeners as it is now, but also is required within the command error handler.
TL;DR: handle `MessageSendForbiddenError` within `on_application_command_error` event listener.

2. The current error capture decorators use the `capture_error_and_close()` function. This should be refactored to determine whether TeX-Bot actually needs closing on a per-decorator basis. The `capture_guild_does_not_exist_error` & `capture_strike_tracking_error` decorators should close TeX-Bot, whereas `capture_message_send_forbidden_error` decorator shouldn't.

**Comment by MattyTheHacker** [comment]
not a clue how to do any of that 🫡
